src/panels/config/automation/condition/ha-automation-condition-editor.ts

```diff
@@ -42,10 +46,21 @@ const OPTIONS = [
 export default class HaAutomationConditionEditor extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @property() public condition!: Condition;
-
   @property() public yamlMode = false;
 
+  private _condition!: Condition;
```

Comment:
Instead of using a getter/setter, let's memoize the output of expandConditionWithShorthand in this class.

Reply:
I don't quite understand. Do we have an example where this is done on a property?

Reply:
Instead of doing it on a property, pass the property to the memoized function inside the `render` method.

```ts
_processCondition = memoizeOne(condition => ...);

render() {
  return html`<ha-something .condition=${this._processCondition(this.condition)}>`;
}
```

Reply:
The benefit is that you can still access the unprocessed condition too. Pushing a commit with the change.